viewer/packages/cad-parsers/src/cad/filterPrimitivesV9.test.ts

```diff
@@ -0,0 +1,365 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import { filterGeometryOutsideClipBox } from './filterPrimitivesV9';
+import { RevealGeometryCollectionType } from '@reveal/sector-parser';
+import * as THREE from 'three';
+
+import {
+  PrimitiveType,
+  Box,
+  Circle,
+  Cone,
+  EccentricCone,
+  Ellipsoid,
+  GeneralCylinder,
+  GeneralRing,
+  Quad,
+  Torus,
+  Trapezium,
+  Nut,
+  getCollectionType
+} from '../../../../test-utilities/src/primitives/primitiveTypes';
+
+import {
+  createPrimitiveInterleavedGeometry,
+  parseInterleavedGeometry
+} from '../../../../test-utilities/src/primitives/primitiveThreeBuffers';
+
+function testSecondFilteredAwayFromBoxAtX10Yneg10Z0(primitives: any[], primitiveType: PrimitiveType) {
+  const bufferGeometry = createPrimitiveInterleavedGeometry(primitiveType, primitives);
+
+  const clipBox = new THREE.Box3(new THREE.Vector3(8, -12, -2), new THREE.Vector3(12, -8, 2));
+
+  const collectionType = getCollectionType(primitiveType);
+
+  const filteredGeometry = filterGeometryOutsideClipBox(bufferGeometry, collectionType, clipBox);
+
+  expect(filteredGeometry).toBeDefined();
+
+  const newPrimitives = parseInterleavedGeometry(primitiveType, filteredGeometry!);
+
+  expect(newPrimitives).toHaveLength(1);
+  expect(newPrimitives[0]).toContainKeys(Object.keys(primitives[0]));
+
+  for (const key in primitives[0]) {
+    if (Array.isArray(primitives[0][key])) {
+      const array0 = primitives[0][key] as number[];
+      const array1 = newPrimitives[0][key] as number[];
+
+      expect(array0).toHaveLength(array1.length);
+
+      for (let i = 0; i < array0.length; i++) {
+        expect(array0[i]).toBeCloseTo(array1[i]);
+      }
+    } else {
+      expect(primitives[0][key]).toBeCloseTo(newPrimitives[0][key] as number);
+    }
+  }
+}
+
+describe('filterPrimitivesV9 filters primitives correctly', () => {
```

Comment:
```suggestion
describe(filterPrimitivesV9.name, () => {
```

A neat trick I recently discovered to ensure tests are not stuck with stale names

Reply:
BTW, test gets 10/10 ⭐

Reply:
`filterPrimitivesV9` is not a class, so I suppose this can't be done?

Reply:
Yes, that will work for functions too